internal/standby/standby.go

```diff
@@ -5,24 +5,36 @@ import (
 	"encoding/json"
 	"fmt"
 	"log/slog"
+	"sync"
 	"time"
 
 	"github.com/EvergenEnergy/remote-standby/internal/config"
 	"github.com/EvergenEnergy/remote-standby/internal/plan"
 	mqtt "github.com/eclipse/paho.mqtt.golang"
 )
 
+type ServiceMode string
+
+const (
+	StandbyMode ServiceMode = "standby"
+	CommandMode ServiceMode = "command"
+)
+
 type Service struct {
-	logger     *slog.Logger
-	cfg        config.Config
-	MQTTClient mqtt.Client
+	logger                *slog.Logger
+	cfg                   config.Config
+	mqttClient            mqtt.Client
+	latestCommandReceived time.Time
+	mode                  ServiceMode
+	mutex                 sync.Mutex
 }
 
 func NewService(logger *slog.Logger, cfg config.Config) *Service {
 	return &Service{
 		logger:     logger,
 		cfg:        cfg,
-		MQTTClient: initMQTTClient(cfg),
+		mqttClient: initMQTTClient(cfg),
```

Comment:
Related to https://github.com/EvergenEnergy/remote-standby/pull/9#discussion_r1604385248: I would rather inject the client interface into `NewService` than create a specific implementation of it here. First of all, for easier testing.

Reply:
thanks, I have created a separate package to create an mqtt client and am injecting it to the standby service